spec/helpers/application_helper_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+# Specs in this file have access to a helper object that includes
+# the ApplicationHelper. For example:
+#
+# describe ApplicationHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe ApplicationHelper, type: :helper do
+  let(:user) { create :user }
+
+  context "user unauthenticated" do
```

Comment:
Each method should have its own context.
```
context '#current_user' do
  ...
end

context '#user_logged_in?' do
  ..
end
```

Also, take a look what's the correct way for context descriptions.

`it 'current ser' do` should be `it 'returns the current user' do`

Reply:
```ruby
  context '#current_user' do
    it 'returns the current ser' do
      session[:user_id] = user.id
      expect(current_user).to eq(user)
    end

    it 'not returns the current user' do
      expect(current_user).to be_nil
    end
  end

  context '#user_logged_in?' do
    it 'user logged in' do
      session[:user_id] = user.id
      expect(user_logged_in?).to eq(user)
    end

    it 'user not logged in' do
      expect(user_logged_in?).to be_nil
    end
  end
```

Thakns for the advise